In [2]:
import pandas as pd

I imported the pandas library and aliased it as 'pd'.

Now I'm going to 'import' the .csv file, so I can view it and work on it here

We can choose to import the .csv file using the full path (procceded by a / with the working directory) or the relative file path. 
The read.csv fucntion is able to decipher between the two, so it doesn't matter. 

In [3]:
df = pd.read_csv('Mental_Health_Care.csv')
df

Indicator              Group  \
0     Took Prescription Medication for Mental Health...  National Estimate   
1     Took Prescription Medication for Mental Health...             By Age   
2     Took Prescription Medication for Mental Health...             By Age   
3     Took Prescription Medication for Mental Health...             By Age   
4     Took Prescription Medication for Mental Health...             By Age   
...                                                 ...                ...   
4779  Needed Counseling or Therapy But Did Not Get I...           By State   
4780  Needed Counseling or Therapy But Did Not Get I...           By State   
4781  Needed Counseling or Therapy But Did Not Get I...           By State   
4782  Needed Counseling or Therapy But Did Not Get I...           By State   
4783  Needed Counseling or Therapy But Did Not Get I...           By State   

              State       Subgroup Phase  Time Period Time Period Label  \
0     United States  United States     2           13   Aug 19 - Aug 31   
1     United States  18 - 29 years     2           13   Aug 19 - Aug 31   
2     United States  30 - 39 years     2           13   Aug 19 - Aug 31   
3     United States  40 - 49 years     2           13   Aug 19 - Aug 31   
4     United States  50 - 59 years     2           13   Aug 19 - Aug 31   
...             ...            ...   ...          ...               ...   
4779       Virginia       Virginia   3.1           28   Apr 14 - Apr 26   
4780     Washington     Washington   3.1           28   Apr 14 - Apr 26   
4781  West Virginia  West Virginia   3.1           28   Apr 14 - Apr 26   
4782      Wisconsin      Wisconsin   3.1           28   Apr 14 - Apr 26   
4783        Wyoming        Wyoming   3.1           28   Apr 14 - Apr 26   

      Time Period Start Date    Time Period End Date  Value  LowCI  HighCI  \
0     08/19/2020 12:00:00 AM  08/31/2020 12:00:00 AM   19.4   19.0    19.8   
1     08/19/2020 12:00:00 AM  08/31/2020 12:00:00 AM   18.7   17.2    20.3   
2     08/19/2020 12:00:00 AM  08/31/2020 12:00:00 AM   18.3   17.3    19.2   
3     08/19/2020 12:00:00 AM  08/31/2020 12:00:00 AM   20.4   19.5    21.3   
4     08/19/2020 12:00:00 AM  08/31/2020 12:00:00 AM   21.2   20.2    22.2   
...                      ...                     ...    ...    ...     ...   
4779  04/14/2021 12:00:00 AM  04/26/2021 12:00:00 AM    8.6    6.5    11.2   
4780  04/14/2021 12:00:00 AM  04/26/2021 12:00:00 AM   11.7    9.5    14.2   
4781  04/14/2021 12:00:00 AM  04/26/2021 12:00:00 AM   13.3    9.5    17.8   
4782  04/14/2021 12:00:00 AM  04/26/2021 12:00:00 AM    7.9    5.6    10.8   
4783  04/14/2021 12:00:00 AM  04/26/2021 12:00:00 AM    6.9    4.7     9.6   

     Confidence Interval Quartile Range  Suppression Flag  
0            19.0 - 19.8            NaN               NaN  
1            17.2 - 20.3            NaN               NaN  
2            17.3 - 19.2            NaN               NaN  
3            19.5 - 21.3            NaN               NaN  
4            20.2 - 22.2            NaN               NaN  
...                  ...            ...               ...  
4779          6.5 - 11.2        8.2-9.7               NaN  
4780          9.5 - 14.2      11.0-14.7               NaN  
4781          9.5 - 17.8      11.0-14.7               NaN  
4782          5.6 - 10.8        5.5-8.1               NaN  
4783           4.7 - 9.6        5.5-8.1               NaN  

[4784 rows x 15 columns]

Scoping the data:
I already see from the first couple of rows that's we have some missing values in our dataset... I want to see how many empty values we actually have with .isnull() or .info()


In [4]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4784 entries, 0 to 4783
Data columns (total 15 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   Indicator               4784 non-null   object 
 1   Group                   4784 non-null   object 
 2   State                   4784 non-null   object 
 3   Subgroup                4784 non-null   object 
 4   Phase                   4784 non-null   object 
 5   Time Period             4784 non-null   int64  
 6   Time Period Label       4784 non-null   object 
 7   Time Period Start Date  4784 non-null   object 
 8   Time Period End Date    4784 non-null   object 
 9   Value                   4609 non-null   float64
 10  LowCI                   4609 non-null   float64
 11  HighCI                  4609 non-null   float64
 12  Confidence Interval     4609 non-null   object 
 13  Quartile Range          3264 non-null   object 
 14  Suppression Flag        7 non-null      

So we can see there's a plethora of missing information. There's supposed to be 4784 entires for each column, but we can see that's not true for some of the columns. We see that there are 175 missing values (4784-4609) in
the columns, Value, LowCI, HighCI(which I'm assuming stands for confidence interval)& Confidence Interval.
-Then there's way more missing in the Quartile range and Suppresion flag.


Data Cleaning -Maintaing Data Integrity: 
So I'm going to be making a new dataframe(a copy of the original one), where I'm free to make changes without messing with the original dataframe.

We'll name this new frame 'df1'. I'll be deleting the null values by dropping all of the columns with null values.

Then after looking at the column names, I want to make a few changes like shorten 'Time Period Start date' and 'Time Period End Date' to just 'Start Date' and 'End Date'.

In [5]:
df1 = df.dropna(axis = 'columns') #dropping columns with null values
#can also be written as df.dropna(axis= 1), will not use inplace = False because that's the default. 
#Nor will I be using inplace = True because not committing changes to the orginal dataset.
df1.shape

(4784, 9)

In [6]:
df1.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4784 entries, 0 to 4783
Data columns (total 9 columns):
 #   Column                  Non-Null Count  Dtype 
---  ------                  --------------  ----- 
 0   Indicator               4784 non-null   object
 1   Group                   4784 non-null   object
 2   State                   4784 non-null   object
 3   Subgroup                4784 non-null   object
 4   Phase                   4784 non-null   object
 5   Time Period             4784 non-null   int64 
 6   Time Period Label       4784 non-null   object
 7   Time Period Start Date  4784 non-null   object
 8   Time Period End Date    4784 non-null   object
dtypes: int64(1), object(8)
memory usage: 336.5+ KB


Here, we can see that we no longer have any null values present. Time to change the column names

In [7]:
df1 = df1.rename(columns={'Time Period Start Date': 'Start Date','Time Period End Date': 'End Date'})
df1


Indicator              Group  \
0     Took Prescription Medication for Mental Health...  National Estimate   
1     Took Prescription Medication for Mental Health...             By Age   
2     Took Prescription Medication for Mental Health...             By Age   
3     Took Prescription Medication for Mental Health...             By Age   
4     Took Prescription Medication for Mental Health...             By Age   
...                                                 ...                ...   
4779  Needed Counseling or Therapy But Did Not Get I...           By State   
4780  Needed Counseling or Therapy But Did Not Get I...           By State   
4781  Needed Counseling or Therapy But Did Not Get I...           By State   
4782  Needed Counseling or Therapy But Did Not Get I...           By State   
4783  Needed Counseling or Therapy But Did Not Get I...           By State   

              State       Subgroup Phase  Time Period Time Period Label  \
0     United States  United States     2           13   Aug 19 - Aug 31   
1     United States  18 - 29 years     2           13   Aug 19 - Aug 31   
2     United States  30 - 39 years     2           13   Aug 19 - Aug 31   
3     United States  40 - 49 years     2           13   Aug 19 - Aug 31   
4     United States  50 - 59 years     2           13   Aug 19 - Aug 31   
...             ...            ...   ...          ...               ...   
4779       Virginia       Virginia   3.1           28   Apr 14 - Apr 26   
4780     Washington     Washington   3.1           28   Apr 14 - Apr 26   
4781  West Virginia  West Virginia   3.1           28   Apr 14 - Apr 26   
4782      Wisconsin      Wisconsin   3.1           28   Apr 14 - Apr 26   
4783        Wyoming        Wyoming   3.1           28   Apr 14 - Apr 26   

                  Start Date                End Date  
0     08/19/2020 12:00:00 AM  08/31/2020 12:00:00 AM  
1     08/19/2020 12:00:00 AM  08/31/2020 12:00:00 AM  
2     08/19/2020 12:00:00 AM  08/31/2020 12:00:00 AM  
3     08/19/2020 12:00:00 AM  08/31/2020 12:00:00 AM  
4     08/19/2020 12:00:00 AM  08/31/2020 12:00:00 AM  
...                      ...                     ...  
4779  04/14/2021 12:00:00 AM  04/26/2021 12:00:00 AM  
4780  04/14/2021 12:00:00 AM  04/26/2021 12:00:00 AM  
4781  04/14/2021 12:00:00 AM  04/26/2021 12:00:00 AM  
4782  04/14/2021 12:00:00 AM  04/26/2021 12:00:00 AM  
4783  04/14/2021 12:00:00 AM  04/26/2021 12:00:00 AM  

[4784 rows x 9 columns]

In [8]:
df1.columns

Index(['Indicator', 'Group', 'State', 'Subgroup', 'Phase', 'Time Period',
       'Time Period Label', 'Start Date', 'End Date'],
      dtype='object')

Okay so we have so many rows that I can't physically scroll all the way down to see all of the unique values of the Indicator column. From looking throught the data, the Indicator column is a defining column of our dataset, so I wan't to see the unique values.
So in order to do that I'll use .unique() for that column, and other columns to just to gage more about my data.

In [9]:
df1.Indicator.unique()

array(['Took Prescription Medication for Mental Health, Last 4 Weeks',
       'Received Counseling or Therapy, Last 4 Weeks',
       'Took Prescription Medication for Mental Health And/Or Received Counseling or Therapy, Last 4 Weeks',
       'Needed Counseling or Therapy But Did Not Get It, Last 4 Weeks'],
      dtype=object)

In [10]:
type(df1.Indicator.head())

pandas.core.series.Series

In [11]:
df1['Indicator'] #we can also use the brackets to pull out a single column, but it's not effective in viewing unique values

0       Took Prescription Medication for Mental Health...
1       Took Prescription Medication for Mental Health...
2       Took Prescription Medication for Mental Health...
3       Took Prescription Medication for Mental Health...
4       Took Prescription Medication for Mental Health...
                              ...                        
4779    Needed Counseling or Therapy But Did Not Get I...
4780    Needed Counseling or Therapy But Did Not Get I...
4781    Needed Counseling or Therapy But Did Not Get I...
4782    Needed Counseling or Therapy But Did Not Get I...
4783    Needed Counseling or Therapy But Did Not Get I...
Name: Indicator, Length: 4784, dtype: object

In [12]:
df1.State.unique()

array(['United States', 'Alabama', 'Alaska', 'Arizona', 'Arkansas',
       'California', 'Colorado', 'Connecticut', 'Delaware',
       'District of Columbia', 'Florida', 'Georgia', 'Hawaii', 'Idaho',
       'Illinois', 'Indiana', 'Iowa', 'Kansas', 'Kentucky', 'Louisiana',
       'Maine', 'Maryland', 'Massachusetts', 'Michigan', 'Minnesota',
       'Mississippi', 'Missouri', 'Montana', 'Nebraska', 'Nevada',
       'New Hampshire', 'New Jersey', 'New Mexico', 'New York',
       'North Carolina', 'North Dakota', 'Ohio', 'Oklahoma', 'Oregon',
       'Pennsylvania', 'Rhode Island', 'South Carolina', 'South Dakota',
       'Tennessee', 'Texas', 'Utah', 'Vermont', 'Virginia', 'Washington',
       'West Virginia', 'Wisconsin', 'Wyoming'], dtype=object)

In [13]:
df1.Group.unique()

array(['National Estimate', 'By Age', 'By Sex',
       'By Presence of Symptoms of Anxiety/Depression',
       'By Race/Hispanic ethnicity', 'By Education', 'By State',
       'By Disability status'], dtype=object)

In [14]:
df1.Subgroup.unique()

array(['United States', '18 - 29 years', '30 - 39 years', '40 - 49 years',
       '50 - 59 years', '60 - 69 years', '70 - 79 years',
       '80 years and above', 'Male', 'Female',
       'Did not experience symptoms of anxiety/depression in the past 4 weeks',
       'Experienced symptoms of anxiety/depression in past 4 weeks',
       'Hispanic or Latino', 'Non-Hispanic White, single race',
       'Non-Hispanic Black, single race',
       'Non-Hispanic Asian, single race',
       'Non-Hispanic, other races and multiple races',
       'Less than a high school diploma', 'High school diploma or GED',
       "Some college/Associate's degree", "Bachelor's degree or higher",
       'Alabama', 'Alaska', 'Arizona', 'Arkansas', 'California',
       'Colorado', 'Connecticut', 'Delaware', 'District of Columbia',
       'Florida', 'Georgia', 'Hawaii', 'Idaho', 'Illinois', 'Indiana',
       'Iowa', 'Kansas', 'Kentucky', 'Louisiana', 'Maine', 'Maryland',
       'Massachusetts', 'Michigan', 'Minnesota

In [15]:
df1[df1.Subgroup == '18 - 29 years']

Indicator   Group  \
1     Took Prescription Medication for Mental Health...  By Age   
73         Received Counseling or Therapy, Last 4 Weeks  By Age   
145   Took Prescription Medication for Mental Health...  By Age   
217   Needed Counseling or Therapy But Did Not Get I...  By Age   
289   Took Prescription Medication for Mental Health...  By Age   
...                                                 ...     ...   
4521       Received Counseling or Therapy, Last 4 Weeks  By Age   
4595  Took Prescription Medication for Mental Health...  By Age   
4616  Took Prescription Medication for Mental Health...  By Age   
4690  Needed Counseling or Therapy But Did Not Get I...  By Age   
4711  Needed Counseling or Therapy But Did Not Get I...  By Age   

              State       Subgroup Phase  Time Period Time Period Label  \
1     United States  18 - 29 years     2           13   Aug 19 - Aug 31   
73    United States  18 - 29 years     2           13   Aug 19 - Aug 31   
145   United States  18 - 29 years     2           13   Aug 19 - Aug 31   
217   United States  18 - 29 years     2           13   Aug 19 - Aug 31   
289   United States  18 - 29 years     2           14    Sep 2 - Sep 14   
...             ...            ...   ...          ...               ...   
4521  United States  18 - 29 years   3.1           28   Apr 14 - Apr 26   
4595  United States  18 - 29 years    -1            1   Mar 30 - Apr 13   
4616  United States  18 - 29 years   3.1           28   Apr 14 - Apr 26   
4690  United States  18 - 29 years    -1            1   Mar 30 - Apr 13   
4711  United States  18 - 29 years   3.1           28   Apr 14 - Apr 26   

                  Start Date                End Date  
1     08/19/2020 12:00:00 AM  08/31/2020 12:00:00 AM  
73    08/19/2020 12:00:00 AM  08/31/2020 12:00:00 AM  
145   08/19/2020 12:00:00 AM  08/31/2020 12:00:00 AM  
217   08/19/2020 12:00:00 AM  08/31/2020 12:00:00 AM  
289   09/02/2020 12:00:00 AM  09/14/2020 12:00:00 AM  
...                      ...                     ...  
4521  04/14/2021 12:00:00 AM  04/26/2021 12:00:00 AM  
4595  03/30/2021 12:00:00 AM  04/13/2021 12:00:00 AM  
4616  04/14/2021 12:00:00 AM  04/26/2021 12:00:00 AM  
4690  03/30/2021 12:00:00 AM  04/13/2021 12:00:00 AM  
4711  04/14/2021 12:00:00 AM  04/26/2021 12:00:00 AM  

[72 rows x 9 columns]

The above code can also be written as df2['Subgroup'] == '18 - 29 years'. It'll also return a bool type output. 
In order to get a non bool type output and the actual values, place [] around the entire filered object code and apply it to your dataframe: df2[df2['Subgroup'] == '18 - 29 years']

In [16]:
df1[df1.State == 'New York']

Indicator     Group     State  \
53    Took Prescription Medication for Mental Health...  By State  New York   
125        Received Counseling or Therapy, Last 4 Weeks  By State  New York   
197   Took Prescription Medication for Mental Health...  By State  New York   
269   Needed Counseling or Therapy But Did Not Get I...  By State  New York   
341   Took Prescription Medication for Mental Health...  By State  New York   
...                                                 ...       ...       ...   
4385  Needed Counseling or Therapy But Did Not Get I...  By State  New York   
4480  Took Prescription Medication for Mental Health...  By State  New York   
4575       Received Counseling or Therapy, Last 4 Weeks  By State  New York   
4670  Took Prescription Medication for Mental Health...  By State  New York   
4765  Needed Counseling or Therapy But Did Not Get I...  By State  New York   

      Subgroup               Phase  Time Period Time Period Label  \
53    New York                   2           13   Aug 19 - Aug 31   
125   New York                   2           13   Aug 19 - Aug 31   
197   New York                   2           13   Aug 19 - Aug 31   
269   New York                   2           13   Aug 19 - Aug 31   
341   New York                   2           14    Sep 2 - Sep 14   
...        ...                 ...          ...               ...   
4385  New York  3 (Jan 6 � Mar 29)           27   Mar 17 - Mar 29   
4480  New York                 3.1           28   Apr 14 - Apr 26   
4575  New York                 3.1           28   Apr 14 - Apr 26   
4670  New York                 3.1           28   Apr 14 - Apr 26   
4765  New York                 3.1           28   Apr 14 - Apr 26   

                  Start Date                End Date  
53    08/19/2020 12:00:00 AM  08/31/2020 12:00:00 AM  
125   08/19/2020 12:00:00 AM  08/31/2020 12:00:00 AM  
197   08/19/2020 12:00:00 AM  08/31/2020 12:00:00 AM  
269   08/19/2020 12:00:00 AM  08/31/2020 12:00:00 AM  
341   09/02/2020 12:00:00 AM  09/14/2020 12:00:00 AM  
...                      ...                     ...  
4385  03/17/2021 12:00:00 AM  03/29/2021 12:00:00 AM  
4480  04/14/2021 12:00:00 AM  04/26/2021 12:00:00 AM  
4575  04/14/2021 12:00:00 AM  04/26/2021 12:00:00 AM  
4670  04/14/2021 12:00:00 AM  04/26/2021 12:00:00 AM  
4765  04/14/2021 12:00:00 AM  04/26/2021 12:00:00 AM  

[64 rows x 9 columns]

So what do I want to filter from this data to be able to analyze? I want to be able to return only specific information of a value in a column. I want to filter the data in which the State has the value 'New York' and the Subgroup has the values is '18-29 years'. 

So i'll be making a variable in order to filter the set. I want to go ahead and put this new filtered set into a seperate dataframe, which I'll call 'df2'. 

In [17]:
df2 = df1[(df1.Subgroup == '18 - 29 years') | (df1.State == 'New York')]
df2
#Similar code df2 = df1.loc[(df1.Subgroup == '18 - 29 years') | (df1.State == 'New York')]
# group = ['18 - 29 years', 'New York'] 
# df2[df2['Subgroup'].isin(group)]

Indicator     Group  \
1     Took Prescription Medication for Mental Health...    By Age   
53    Took Prescription Medication for Mental Health...  By State   
73         Received Counseling or Therapy, Last 4 Weeks    By Age   
125        Received Counseling or Therapy, Last 4 Weeks  By State   
145   Took Prescription Medication for Mental Health...    By Age   
...                                                 ...       ...   
4616  Took Prescription Medication for Mental Health...    By Age   
4670  Took Prescription Medication for Mental Health...  By State   
4690  Needed Counseling or Therapy But Did Not Get I...    By Age   
4711  Needed Counseling or Therapy But Did Not Get I...    By Age   
4765  Needed Counseling or Therapy But Did Not Get I...  By State   

              State       Subgroup Phase  Time Period Time Period Label  \
1     United States  18 - 29 years     2           13   Aug 19 - Aug 31   
53         New York       New York     2           13   Aug 19 - Aug 31   
73    United States  18 - 29 years     2           13   Aug 19 - Aug 31   
125        New York       New York     2           13   Aug 19 - Aug 31   
145   United States  18 - 29 years     2           13   Aug 19 - Aug 31   
...             ...            ...   ...          ...               ...   
4616  United States  18 - 29 years   3.1           28   Apr 14 - Apr 26   
4670       New York       New York   3.1           28   Apr 14 - Apr 26   
4690  United States  18 - 29 years    -1            1   Mar 30 - Apr 13   
4711  United States  18 - 29 years   3.1           28   Apr 14 - Apr 26   
4765       New York       New York   3.1           28   Apr 14 - Apr 26   

                  Start Date                End Date  
1     08/19/2020 12:00:00 AM  08/31/2020 12:00:00 AM  
53    08/19/2020 12:00:00 AM  08/31/2020 12:00:00 AM  
73    08/19/2020 12:00:00 AM  08/31/2020 12:00:00 AM  
125   08/19/2020 12:00:00 AM  08/31/2020 12:00:00 AM  
145   08/19/2020 12:00:00 AM  08/31/2020 12:00:00 AM  
...                      ...                     ...  
4616  04/14/2021 12:00:00 AM  04/26/2021 12:00:00 AM  
4670  04/14/2021 12:00:00 AM  04/26/2021 12:00:00 AM  
4690  03/30/2021 12:00:00 AM  04/13/2021 12:00:00 AM  
4711  04/14/2021 12:00:00 AM  04/26/2021 12:00:00 AM  
4765  04/14/2021 12:00:00 AM  04/26/2021 12:00:00 AM  

[136 rows x 9 columns]

In order to filter two values, you need to create a list. So I'm going to create a variable that contains the lists of filtered values, then use " .isin() " to apply it to my filtering code. Pandas isin() is used to filter data frames and selects the rows that have particular values in a given column.

In [18]:
group = ['18 - 29 years', 'New York'] 
df2[df2['Subgroup'].isin(group)]

Indicator     Group  \
1     Took Prescription Medication for Mental Health...    By Age   
53    Took Prescription Medication for Mental Health...  By State   
73         Received Counseling or Therapy, Last 4 Weeks    By Age   
125        Received Counseling or Therapy, Last 4 Weeks  By State   
145   Took Prescription Medication for Mental Health...    By Age   
...                                                 ...       ...   
4616  Took Prescription Medication for Mental Health...    By Age   
4670  Took Prescription Medication for Mental Health...  By State   
4690  Needed Counseling or Therapy But Did Not Get I...    By Age   
4711  Needed Counseling or Therapy But Did Not Get I...    By Age   
4765  Needed Counseling or Therapy But Did Not Get I...  By State   

              State       Subgroup Phase  Time Period Time Period Label  \
1     United States  18 - 29 years     2           13   Aug 19 - Aug 31   
53         New York       New York     2           13   Aug 19 - Aug 31   
73    United States  18 - 29 years     2           13   Aug 19 - Aug 31   
125        New York       New York     2           13   Aug 19 - Aug 31   
145   United States  18 - 29 years     2           13   Aug 19 - Aug 31   
...             ...            ...   ...          ...               ...   
4616  United States  18 - 29 years   3.1           28   Apr 14 - Apr 26   
4670       New York       New York   3.1           28   Apr 14 - Apr 26   
4690  United States  18 - 29 years    -1            1   Mar 30 - Apr 13   
4711  United States  18 - 29 years   3.1           28   Apr 14 - Apr 26   
4765       New York       New York   3.1           28   Apr 14 - Apr 26   

                  Start Date                End Date  
1     08/19/2020 12:00:00 AM  08/31/2020 12:00:00 AM  
53    08/19/2020 12:00:00 AM  08/31/2020 12:00:00 AM  
73    08/19/2020 12:00:00 AM  08/31/2020 12:00:00 AM  
125   08/19/2020 12:00:00 AM  08/31/2020 12:00:00 AM  
145   08/19/2020 12:00:00 AM  08/31/2020 12:00:00 AM  
...                      ...                     ...  
4616  04/14/2021 12:00:00 AM  04/26/2021 12:00:00 AM  
4670  04/14/2021 12:00:00 AM  04/26/2021 12:00:00 AM  
4690  03/30/2021 12:00:00 AM  04/13/2021 12:00:00 AM  
4711  04/14/2021 12:00:00 AM  04/26/2021 12:00:00 AM  
4765  04/14/2021 12:00:00 AM  04/26/2021 12:00:00 AM  

[136 rows x 9 columns]

the .loc() fucntions allows you to filter out specific columns and also to add particular parameters to the filter.
For example to filter out columns for the data to be more concise, like what we did above. 
We used the .loc() function to pass the two columns as the parameters that we wanted to see data in.



In [19]:
df2.loc[(df.Phase >= '2'),['Indicator','Phase']]

Indicator Phase
1     Took Prescription Medication for Mental Health...     2
53    Took Prescription Medication for Mental Health...     2
73         Received Counseling or Therapy, Last 4 Weeks     2
125        Received Counseling or Therapy, Last 4 Weeks     2
145   Took Prescription Medication for Mental Health...     2
...                                                 ...   ...
4575       Received Counseling or Therapy, Last 4 Weeks   3.1
4616  Took Prescription Medication for Mental Health...   3.1
4670  Took Prescription Medication for Mental Health...   3.1
4711  Needed Counseling or Therapy But Did Not Get I...   3.1
4765  Needed Counseling or Therapy But Did Not Get I...   3.1

[128 rows x 2 columns]

In [20]:
#We want to return Indicator and Subgroup information, where the subgroup is 18-29 years
df2.loc[(df2.Subgroup == '18 - 29 years'),['Indicator','Subgroup']].head()



Indicator       Subgroup
1    Took Prescription Medication for Mental Health...  18 - 29 years
73        Received Counseling or Therapy, Last 4 Weeks  18 - 29 years
145  Took Prescription Medication for Mental Health...  18 - 29 years
217  Needed Counseling or Therapy But Did Not Get I...  18 - 29 years
289  Took Prescription Medication for Mental Health...  18 - 29 years

In [21]:
df2.loc[df2['Subgroup'].isin(['18 - 29 years', 'New York']),['Indicator','Subgroup']].head()

Indicator       Subgroup
1    Took Prescription Medication for Mental Health...  18 - 29 years
53   Took Prescription Medication for Mental Health...       New York
73        Received Counseling or Therapy, Last 4 Weeks  18 - 29 years
125       Received Counseling or Therapy, Last 4 Weeks       New York
145  Took Prescription Medication for Mental Health...  18 - 29 years

In [22]:
group = ['18 - 29 years', 'New York']
df2.loc[df2['Subgroup'].isin(group),['Indicator','Subgroup']].head()

Indicator       Subgroup
1    Took Prescription Medication for Mental Health...  18 - 29 years
53   Took Prescription Medication for Mental Health...       New York
73        Received Counseling or Therapy, Last 4 Weeks  18 - 29 years
125       Received Counseling or Therapy, Last 4 Weeks       New York
145  Took Prescription Medication for Mental Health...  18 - 29 years

In [23]:
deleted = ['Start Date', 'End Date', 'Time Period', 'Phase']
df3 = df2.drop(deleted, axis = 1)
df3

#other ways to write this line of code
# df3 = df2.drop(['Start Date', 'End Date'], 1)
# df3 = df2.drop(columns = ['Start Date','End Date'],axis = 1)
# df3 = df2.drop(['Start Date', 'End Date'], axis = 1)

Indicator     Group  \
1     Took Prescription Medication for Mental Health...    By Age   
53    Took Prescription Medication for Mental Health...  By State   
73         Received Counseling or Therapy, Last 4 Weeks    By Age   
125        Received Counseling or Therapy, Last 4 Weeks  By State   
145   Took Prescription Medication for Mental Health...    By Age   
...                                                 ...       ...   
4616  Took Prescription Medication for Mental Health...    By Age   
4670  Took Prescription Medication for Mental Health...  By State   
4690  Needed Counseling or Therapy But Did Not Get I...    By Age   
4711  Needed Counseling or Therapy But Did Not Get I...    By Age   
4765  Needed Counseling or Therapy But Did Not Get I...  By State   

              State       Subgroup Time Period Label  
1     United States  18 - 29 years   Aug 19 - Aug 31  
53         New York       New York   Aug 19 - Aug 31  
73    United States  18 - 29 years   Aug 19 - Aug 31  
125        New York       New York   Aug 19 - Aug 31  
145   United States  18 - 29 years   Aug 19 - Aug 31  
...             ...            ...               ...  
4616  United States  18 - 29 years   Apr 14 - Apr 26  
4670       New York       New York   Apr 14 - Apr 26  
4690  United States  18 - 29 years   Mar 30 - Apr 13  
4711  United States  18 - 29 years   Apr 14 - Apr 26  
4765       New York       New York   Apr 14 - Apr 26  

[136 rows x 5 columns]

In [24]:
df3

Indicator     Group  \
1     Took Prescription Medication for Mental Health...    By Age   
53    Took Prescription Medication for Mental Health...  By State   
73         Received Counseling or Therapy, Last 4 Weeks    By Age   
125        Received Counseling or Therapy, Last 4 Weeks  By State   
145   Took Prescription Medication for Mental Health...    By Age   
...                                                 ...       ...   
4616  Took Prescription Medication for Mental Health...    By Age   
4670  Took Prescription Medication for Mental Health...  By State   
4690  Needed Counseling or Therapy But Did Not Get I...    By Age   
4711  Needed Counseling or Therapy But Did Not Get I...    By Age   
4765  Needed Counseling or Therapy But Did Not Get I...  By State   

              State       Subgroup Time Period Label  
1     United States  18 - 29 years   Aug 19 - Aug 31  
53         New York       New York   Aug 19 - Aug 31  
73    United States  18 - 29 years   Aug 19 - Aug 31  
125        New York       New York   Aug 19 - Aug 31  
145   United States  18 - 29 years   Aug 19 - Aug 31  
...             ...            ...               ...  
4616  United States  18 - 29 years   Apr 14 - Apr 26  
4670       New York       New York   Apr 14 - Apr 26  
4690  United States  18 - 29 years   Mar 30 - Apr 13  
4711  United States  18 - 29 years   Apr 14 - Apr 26  
4765       New York       New York   Apr 14 - Apr 26  

[136 rows x 5 columns]

In [25]:
df3.sort_values(by = ['Indicator', 'Subgroup'], inplace = True)
df3

Indicator     Group  \
217   Needed Counseling or Therapy But Did Not Get I...    By Age   
505   Needed Counseling or Therapy But Did Not Get I...    By Age   
793   Needed Counseling or Therapy But Did Not Get I...    By Age   
1081  Needed Counseling or Therapy But Did Not Get I...    By Age   
1369  Needed Counseling or Therapy But Did Not Get I...    By Age   
...                                                 ...       ...   
3305  Took Prescription Medication for Mental Health...  By State   
3593  Took Prescription Medication for Mental Health...  By State   
3881  Took Prescription Medication for Mental Health...  By State   
4169  Took Prescription Medication for Mental Health...  By State   
4480  Took Prescription Medication for Mental Health...  By State   

              State       Subgroup Time Period Label  
217   United States  18 - 29 years   Aug 19 - Aug 31  
505   United States  18 - 29 years    Sep 2 - Sep 14  
793   United States  18 - 29 years   Sep 16 - Sep 28  
1081  United States  18 - 29 years   Sep 30 - Oct 12  
1369  United States  18 - 29 years   Oct 14 - Oct 26  
...             ...            ...               ...  
3305       New York       New York    Feb 3 - Feb 15  
3593       New York       New York    Feb 17 - Mar 1  
3881       New York       New York    Mar 3 - Mar 15  
4169       New York       New York   Mar 17 - Mar 29  
4480       New York       New York   Apr 14 - Apr 26  

[136 rows x 5 columns]